frontend/src/workspace/components/plotter.rs

```diff
@@ -38,39 +42,54 @@ impl Component for Plotter {
     fn change(&mut self, props: Self::Properties) -> ShouldRender {
         self.props = props;
 
-        use plotters::prelude::*;
-
-        if let Some(canvas) = self.canvas.cast::<HtmlCanvasElement>() {
-            let backend = CanvasBackend::with_canvas_object(canvas).unwrap();
-            let root = backend.into_drawing_area();
-            root.fill(&WHITE).unwrap();
-
-            let mut chart = ChartBuilder::on(&root)
-                .x_label_area_size(30)
-                .y_label_area_size(30)
-                .build_ranged(0f32..440., -1f32..1f32).unwrap();
-            chart.configure_mesh().x_labels(3).y_labels(3).draw().unwrap();
-
-            let colors = [BLUE, GREEN];
-
-            for (i, input) in self.props.indication.inputs.iter().enumerate() {
-                if let Some(input) = input {
-                    let series = input
-                        .iter()
-                        .enumerate()
-                        .map(|(x, y)| (x as f32, *y))
-                        .collect::<Vec<(f32, f32)>>();
-                    chart.draw_series(LineSeries::new(series, &colors[i])).unwrap();
+        if let Some(input) = &self.props.indication.inputs[0] {
+            let (channel_1, channel_2): (Vec<f32>, Vec<f32>) = input.into_iter().enumerate().partition_map(|(i, sample)| {
+                if i % 2 == 0 {
+                    Either::Left(sample)
+                } else {
+                    Either::Right(sample)
```

Comment:
This makes me extra happy if the `Either::Left` is the left channel lol. `Either` for an actual binary option beyond just quasi result types (*cough* haskell) is a nice touch.